iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -450,7 +452,9 @@ public boolean isValidMetadataTable(String metaTableName) {
   public URI getURIForAuth(org.apache.hadoop.hive.metastore.api.Table hmsTable) throws URISyntaxException {
     String dbName = hmsTable.getDbName();
     String tableName = hmsTable.getTableName();
-    return new URI(ICEBERG_URI_PREFIX + dbName + "/" + tableName);
+    Table table = IcebergTableUtil.getTable(conf, hmsTable);
```

Comment:
We ran into problems with the approach of loading the Iceberg table here before. The problem is that this method can be called to authorize CREATE TABLE commands as well, at which point the iceberg table does not exist yet, so this will lead to NPE.

If the table object is null, then maybe we can use the hmsTable.getSd().location() + "/metadata"? I'm not sure though, just thinking out loud

Reply:
Giving back `hmsTable.getSd().location() + "/metadata"` seems reasonable to me in such cases

Reply:
What happens if we provide the `metadata_location` in the table properties when creating the table?
Or do we plan to handle this in a different PR?

Reply:
Good point, I will add a change to address this use case as well.